### PR TITLE
Extend sparse linear algebra interface with multiply-add

### PR DESCRIPTION
**Pull request by l90lpa**
This PR extends Atlas's sparse linear algebra interface with multiply-add functionality.

### CODE REVIEW

**Comment by l90lpa** [comment]
Removed dependency on PR https://github.com/ecmwf/atlas/pull/237

**Comment by l90lpa** [comment]
Hi @wdeconinck, this PR currently only exposes the simple $y=Ax+y$ interface. I'm wondering if you have an actual need for $y=\alpha Ax+\beta y$ and if not whether we could leave this as follow-on action?

**Comment by wdeconinck** [comment]
Thanks @l90lpa for this PR. I will review it.

> Hi @wdeconinck, this PR currently only exposes the simple y = A x + y interface. I'm wondering if you have an actual need for y = α A x + β y and if not whether we could leave this as follow-on action?

Hi @l90lpa I think it's OK indeed to leave it as a follow-on action if ever needed, but as you say, unlikely.

File: src/atlas/interpolation/method/Method.cc
Comment (wdeconinck) [inline src/atlas/interpolation/method/Method.cc:1]: Nice cleanup with new API

**Review by wdeconinck — APPROVED** [review]
I think it's really good and cleans up quite a nice chunk in the adjoint code path, and will improve performance! This could be merged as is a.f.a.i.c

**Pull request merged**